src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -66,44 +64,16 @@
 # Docker image tags for the tools
 DEFAULT_IMG_TAG = 'latest'
 
-# directory containing the TaskFarmer job scripts under the root directory
-TASKFARMER_JOB_DIR = 'task_farmer_jobs'
-TASK_INFO_FILE = 'task_info.jsonl'  # file containing the information of each task
-
-
-def _run_command(command, job_dir, log_file_prefix='', check_return_code=True):
-    """
-    Runs the specified command and captures its standard output and standard error.
-    """
-    log_dir = os.path.join(job_dir, 'logs')
-    os.makedirs(log_dir, exist_ok=True)
-    std_out_file = os.path.join(log_dir,
-                                f'stdout_{log_file_prefix}_{datetime.datetime.now().strftime("%Y%m%d_%H%M%S")}')
-    std_err_file = os.path.join(log_dir,
-                                f'stderr_{log_file_prefix}_{datetime.datetime.now().strftime("%Y%m%d_%H%M%S")}')
-
-    with open(std_out_file, "w") as std_out, open(std_err_file, "w") as std_err:
-        p = subprocess.Popen(command, stdout=std_out, stderr=std_err, text=True)
-
-    exit_code = p.wait()
-
-    if check_return_code and exit_code != 0:
-        with open(std_out_file, "r") as std_out, open(std_err_file, "r") as std_err:
-            raise ValueError(f"Error running command '{command}'.\n"
-                             f"Standard output: {std_out.read()}\n"
-                             f"Standard error: {std_err.read()}")
-
-    return std_out_file, std_err_file, exit_code
-
 
 def _pull_image(image_str, job_dir):
     """
     Pulls the specified Shifter image from the specified registry.
     """
 
     print(f"Fetching Shifter image {image_str}...")
-    sp_std_out_file, sp_std_err_file, sp_exit_code = _run_command(["shifterimg", "pull", image_str], job_dir,
-                                                                  log_file_prefix='shifterimg_pull')
+    sp_std_out_file, sp_std_err_file, sp_exit_code = tf_common.run_nersc_command(["shifterimg", "pull", image_str],
+                                                                                 job_dir,
+                                                                                 log_file_prefix='shifterimg_pull')
```

Comment:
```suggestion
    sp_std_out_file, sp_std_err_file, sp_exit_code = tf_common.run_nersc_command(
        ["shifterimg", "pull", image_str], job_dir, log_file_prefix='shifterimg_pull')
```

Reply:
👍

Reply:
There's a few other places in the file that have the squished up command lines, can you fix those?

Reply:
👍

Reply:
:+1: